cosmwasm/its.ts

```diff
@@ -0,0 +1,172 @@
+        },
+    };
+
+    const [account] = await client.accounts;
+    printInfo('Registering token ', JSON.stringify(msg.register_p2p_token_instance));
+
+    if (!dryRun) {
+        await client.execute(account.address, interchainTokenServiceAddress, msg, 'auto');
+    }
+}
+
+export async function checkSingleTokenRegistration(
+    client: CosmWasmClient,
+    interchainTokenServiceAddress: string,
+    tokenId: string,
+    axelarChainId: string,
+): Promise<boolean> {
+    const registered = await client.queryContractSmart(interchainTokenServiceAddress, {
+        token_instance: { chain: axelarChainId, token_id: formatTokenAddress(tokenId) },
+    });
+    return registered;
+}
+
+function formatTokenAddress(tokenAddress: string): string {
+    if (tokenAddress.startsWith('0x')) {
+        return tokenAddress.slice(2);
+    }
+    return tokenAddress;
+}
+
+async function registerSingleToken(client: ClientManager, config: ConfigManager, options) {
+    const { chain, tokenId, originChain, decimals, supply, dryRun } = options;
+    try {
+        const tokenDataToRegister = {
+            tokenId: tokenId,
+            originChain: originChain,
+            decimals: decimals,
+            supply: supply,
+            axelarId: config.getChainConfig(chain).axelarId,
+        };
+        const interchainTokenServiceAddress = config.getContractConfig('InterchainTokenService').address;
+
+        if (!interchainTokenServiceAddress) {
+            throw new Error('InterchainTokenService contract address not found');
+        }
+        await registerToken(interchainTokenServiceAddress, client, tokenDataToRegister, dryRun);
+        printInfo(`Token ${tokenId} on ${chain} is registered successfully`);
+    } catch (e) {
+        printError(`Error registering token ${tokenId} on ${chain}: ${e}`);
+    }
+}
+
+async function checkTokensRegistration(client: CosmWasmClient, config: ConfigManager, options) {
+    const { chains, tokenIds } = options;
+
+    try {
+        const interchainTokenServiceAddress = config.getContractConfig('InterchainTokenService').address;
+
+        if (!interchainTokenServiceAddress) {
+            throw new Error('InterchainTokenService contract address not found');
+        }
+
+        await Promise.all(
+            tokenIds.flatMap((tokenId) => {
+                return chains.map(async (chainName) => {
+                    try {
+                        const registered = await checkSingleTokenRegistration(
+                            client,
+                            interchainTokenServiceAddress,
+                            tokenId,
+                            config.getChainConfig(chainName).axelarId,
+                        );
+                        printInfo(`Token ${tokenId} on ${chainName} is ${registered ? 'registered' : 'not registered'}`);
+                    } catch (e) {
+                        printError(`Error checking token ${tokenId} on ${chainName}: ${e}`);
```

Comment:
Avoid try/catch over other approaches? If the CLI is not maintained, then share your feedback